### PR TITLE
parse buffer from different bases and data unit

### PR DESCRIPTION
**Pull request by urbanishimwe**
All flags that expect buffer as input i.e. `--output-file-size-limit`, `--output-file-max-size-limit`, `--copy-buffer-size` and `input-raw-buffer-size` can now parse  inputs from differents bases and data units like: `10mb`, `10kb`, `100gb`,   `18tb`, `11839023`.... 
data units and bases are case insensitive, the parser accepts only the format of [Go integer literals](https://golang.org/ref/spec#Integer_literals)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ FADDR = :8000
 RUN = docker run -v `pwd`:$(SOURCE_PATH) -p 0.0.0.0:$(PORT):$(PORT) -i -t gor
 BENCHMARK = BenchmarkRAWInput
 TEST = TestRawListenerBench
-VERSION = DEV-$(shell date +%s)
+VERSION = DEV-$(shell grep -oP "[\d.]+" version.go)
```
[the remark this answers is not in the document]
Comment (urbanishimwe) [reply Makefile:8]: sounds more reasonable!

File: settings.go [hunk 2]
```diff
@@ -210,19 +237,70 @@ func init() {
...
-var previousDebugTime int64
-var debugMutex sync.Mutex
-
-// Debug gets called only if --verbose flag specified
+// Debug take an effect only if --verbose flag specified
 func Debug(args ...interface{}) {
 	if Settings.verbose {
-		debugMutex.Lock()
 		now := time.Now()
-		diff := float64(now.UnixNano()-previousDebugTime) / 1000000
```
[the remark this answers is not in the document]
Comment (urbanishimwe) [reply settings.go:221]: Printing diff in `Nanoseconds`! I don't think it's very helpful in the user's perspective. If you can see only the current time(well parsed) of a debug, it's very enough to take relevant action.

Why do you think? Cc: @buger

[the remark this answers is not in the document]
Comment (buger) [reply settings.go:221]: The thing is that the Debug function was doing 2 different things. First is debug output, and the second is performance measurement. So you can wrap your code into 2 Debug calls, and can see how much time it took to run.

File: settings.go [hunk 3]
```diff
@@ -210,19 +238,78 @@ func init() {
...
 	if Settings.verbose {
 		debugMutex.Lock()
+		defer debugMutex.Unlock()
 		now := time.Now()
-		diff := float64(now.UnixNano()-previousDebugTime) / 1000000
-		previousDebugTime = now.UnixNano()
-		debugMutex.Unlock()
-
-		fmt.Printf("[DEBUG][PID %d][%d][%fms] ", os.Getpid(), now.UnixNano(), diff)
+		diff := now.Sub(previousDebugTime).String()
```
Comment (buger) [inline settings.go:251]: If you leave it as "ns" then in the debug call on line 253 at least say that it is `ns`

Comment (urbanishimwe) [reply settings.go:251]: the built in `Duration.String()` format well by giving it relevant prefix

**Pull request merged**